src/pages/journal/journal.ts

```diff
@@ -171,6 +178,64 @@ export class JournalPage extends BasePageComponent implements OnInit {
     }
   }
 
+  /**
+   * If state contains no cached data then get users search results
+   * otherwise get create slot data directly
+   * @param emission
+   */
+  generateSlotAndSearchResults = (emission: SlotItem[]) => {
+    const noCacheData: boolean = true;
+    if (noCacheData && !this.searchResultsAppRefs) {
+      return this.getSearchResults(emission);
+    }
+    this.createSlots(emission, this.searchResultsAppRefs);
+
+  }
+
+  /**
+   * Call endpoint to obtain users previously completed tests and pass to createSlots function
+   * for evaluation
+   * @param emission
+   */
+  getSearchResults = (emission: SlotItem[]) => {
+    const advancedSearchParams: AdvancedSearchParams = {
+      startDate: '',
+      endDate: '',
+      staffNumber: removeLeadingZeros('1234567'),
+      costCode: '',
+    };
+    forkJoin([
+      this.searchProvider.advancedSearch(advancedSearchParams),
+    ]).subscribe(
+      ([
+         searchResultsTemp,
+       ]) => {
+        this.searchResultsAppRefs = searchResultsTemp.map((res) => {
+          return res.applicationReference;
+        });
+        this.createSlots(emission, this.searchResultsAppRefs);
+      },
+      err => this.showError(err),
+    );
+  }
+
+  /**
+   * Search for slots appReference in returned search results
+   * @param slot
+   * @param references
+   */
+  hasSlotBeenTested(slot, references): boolean {
+    const applicationReference: ApplicationReference = {
+      applicationId: slot.slotData.booking.application.applicationId,
+      bookingSequence: slot.slotData.booking.application.bookingSequence,
+      checkDigit: slot.slotData.booking.application.checkDigit,
+    };
+
+    return references.some((reference) => {
+      return reference === parseInt(formatApplicationReference(applicationReference), 10);
+    });
+  }
+
```

Comment:
Perhaps you were already planning on this, but I think the work to retrieve the app refs should be in a separate provider to keep things clean. The API for the provider could be -

- Takes a startDate, endDate and staffNumber
- Returns a list of completed app refs

The journal page would only then be responsible for the business logic of matching completed app refs to the journal bookings.

Reply:
fully agree. will look to refactor.

Reply:
This functionality has been moved to the journal effects now